pkg/cri/annotations/annotations.go

```diff
@@ -59,4 +59,7 @@ const (
 
 	// PodAnnotations are the annotations of the pod
 	PodAnnotations = "io.kubernetes.cri.pod-annotations"
+
+	// WindowsHostProcess is used by hcsshim to identify windows pods that are running HostProcesses
+	WindowsHostProcess = "microsoft.com/hostprocess-container"
```

Comment:
Should it be `com.microsoft.xxx` as like other annotations?

Reply:
This annotation was following the Kubernetes guidelines https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set. This is what it's set to/k8s checks against so there's not much we could do here at the moment in terms of changing it. As of right now in 1.22, Kubernetes is setting this annotation explicitly if the HostProcess field is filled in so that containerd doesn't need to be aware of the cri fields (even though that's what this PR is adding now that the 1.22 deps are vendored in 😆). In 1.23 of k8s I believe they're removing this explicit addition of the annotation if this feature goes to beta so we can change the annotation name then, as the only other change would just be renaming it in our windows containerd-shim to match once k8s no longer needs to care how it's named.

Reply:
Oh, I didn't know that.

https://github.com/kubernetes/kubernetes/blob/7705b300e2085c3864bb1e49a7302bf17f080219/pkg/kubelet/kuberuntime/labels.go#L46-L50

Reply:
The situation is actually a bit weird right now. k8s states annotations should be in the form `foo.com/bar-baz`, where [OCI](https://github.com/opencontainers/runtime-spec/blob/master/config.md#annotations) states they should be in the form `com.foo.bar-baz`.

Probably the most technically correct thing to do would be for the CRI plugin to translate from the k8s form to the OCI form, but it doesn't do that today.